src/runtime/runtime.ts

```diff
@@ -127,4 +132,65 @@ export class Runtime<DependenciesSnakeT, DependenciesCamelT> {
 			},
 		});
 	}
+
+	/**
+	 * Only runs on a CORS preflight request— returns a response with the
+	 * appropriate CORS headers & status.
+	 * 
+	 * @param req The preflight OPTIONS request
+	 * @returns The full response to the preflight request
+	 */
+	public corsPreflight(req: Request): Response {
+		const origin = req.headers.get("Origin");
+		if (origin) {
+			const normalizedOrigin = new URL(origin).origin;
+			if (this.config.cors) {
+				if (this.config.cors.origins.has(normalizedOrigin)) {
+					return new Response(undefined, {
+						status: 204,
+						headers: {
+							...this.corsHeaders(req),
+							"Vary": "Origin",
+						},
+					});
+				}
+			}
+		}
+
+		// Origin is not allowed/no origin header on preflight
+		return new Response(undefined, {
+			status: 403,
+			headers: {
+				"Vary": "Origin",
+				"See": "https://opengb.dev/docs/cors",
```

Comment:
i'm fairly certain OPTIONS can have a response body. let's return an error body that matches our error response format. instead of this header.

if not, then use a header like `X-Error-Documentation`